config/config.go

```diff
@@ -180,14 +181,22 @@ func load(reader io.Reader) (*Config, error) {
 		case name == "generate":
 			gen, err = handleGenerate(s)
 		case strings.HasPrefix(name, "generate"):
+			// Check to see if we have the shorten version of a generate config:
+			// [generate js].
 			sParts := strings.Split(name, " ")
 			if len(sParts) != 2 {
 				return nil, fmt.Errorf("generate section name should have 2 values, not %d", len(sParts))
 			}
-			// Check to see if we have the shorten version of a generate config:
-			// [generate js].
+
 			gen, err = handleGenerate(s)
-			gen.ProtocGen = strings.Trim(sParts[1], "\"")
+			generator := strings.Trim(sParts[1], "\"")
+			// Is this shortened generator a protoc-gen-* binary, or
+			// should it be passed to protoc.
+			if _, err := exec.LookPath("protoc-gen-" + generator); err == nil {
+				gen.Command = "protoc-gen-" + generator
```

Comment:
why not use the found path directly, since we do the work for that here already?

the only disadvantage I can see is that if we support `-x` for generators, then we'd show `/bin/some-generator flags...` instead of `some-generator flags...`. If that's the reason, seems fine to me to keep it as it is, but I'd add a comment to prevent others from slightly optimising the code like I suggested.

Reply:
We can use the path directly, but I wanted to keep it the same as the normal generate section that doesn't lookup the path. If we wanted to use the path we should also do the same for the normal generate section.

Reply:
True. Please add a short comment :)

Reply:
Done.